added option to get decimal point in uncertainty formatted with "parenthesis" notation

alusiani opened the pull request:
Hi,

I propose a small unobstructive addition to the code in order to optionally get the decimal point formatted when using the "parenthesis" notation.  Example of usage:

```
format(set_errors(123.456, 12.345), notation = "parenthesis", digits=4)
## -> '123.46(1234)'
## default formatting
```

```
saved_options = options(errors.parenthesis.unc.dec.point = TRUE)
format(set_errors(123.456, 12.345), notation = "parenthesis", digits=4)
options(saved_options)
## -> '123.46(12.34)'
## optional formatting
```

Enchufa2 commented on R/print.R:
```diff
@@ -34,21 +37,22 @@ format.errors = function(x,
                          ...)
 {
   stopifnot(notation %in% c("parenthesis", "plus-minus"))
-
```
Please do not remove this space

Enchufa2 commented on R/print.R:
```diff
@@ -34,21 +37,22 @@ format.errors = function(x,
   if (is.null(digits))
     digits <- getOption("errors.digits", 1)
   digits <- if (digits == "pdg") digits_pdg(.e(x)) else rep(digits, length(x))
-
```
Please do not remove this space

Enchufa2 commented on R/print.R:
```diff
@@ -34,21 +37,22 @@ format.errors = function(x,
   nulle <- e == 0 & !is.na(e)
-  xexp <- ifelse(.v(x) == 0, get_exponent(e) + 1, get_exponent(x))
-  value_digits <- ifelse(e, digits - get_exponent(e), digits)
+  e_expon = get_exponent(e)
```
Please, call it `eexp` and use the `<-` operator to match the style of the rest.

Enchufa2 commented on R/print.R:
```diff
@@ -34,21 +37,22 @@ format.errors = function(x,
-  xexp <- ifelse(.v(x) == 0, get_exponent(e) + 1, get_exponent(x))
-  value_digits <- ifelse(e, digits - get_exponent(e), digits)
+  e_expon = get_exponent(e)
+  xexp <- ifelse(.v(x)==0, e_expon+1, get_exponent(x))
```
Please add spaces around the operators.

Enchufa2 commented on R/print.R:
```diff
@@ -34,21 +37,22 @@ format.errors = function(x,
+  xexp <- ifelse(.v(x)==0, e_expon+1, get_exponent(x))
+  value_digits <- ifelse(e, digits - e_expon, digits)
   value <- ifelse(e, signif(.v(x), xexp + value_digits), .v(x))
+  ##+++ str(value)
```
Remove

Enchufa2 commented on R/print.R:
```diff
@@ -34,21 +37,22 @@ format.errors = function(x,
   value <- ifelse(e, signif(.v(x), xexp + value_digits), .v(x))
+  ##+++ str(value)
   value <- ifelse(is.finite(value), value, .v(x))
+  ##+++ str(value)
```
Remove

Enchufa2 commented on R/print.R:
```diff
@@ -81,7 +92,6 @@ format.errors = function(x,
             decimal.mark=getOption("OutDec"))
   })
   e <- sub("\\.$", "", e)
-
```
Please restore

Enchufa2 commented on tests/testthat/test-print.R:
```diff
@@ -17,9 +17,6 @@ test_that("error formatting works properly", {
   expect_equal(format(x, notation="parenthesis", digits=3),
                c("10000(12300000)", "11110(1230)", "11111.2(123)", "11111.22(123)",
                  "11111.222(123)", "11111.2222(123)", "11111.2222200(123)", "11111.2222200000(123)"))
-  expect_equal(format(x, notation="parenthesis", digits="pdg"),
-               c("10000(12000000)", "11100(1200)", "11111(12)", "11111.2(12)",
-                 "11111.22(12)", "11111.222(12)", "11111.222220(12)", "11111.222220000(12)"))
```
Please restore

Enchufa2 commented on tests/testthat/test-print.R:
```diff
@@ -32,33 +29,27 @@ test_that("error formatting works properly", {
     c("10000", "12300000"), c("11110", "1230"), c("11111.2", "12.3"), c("11111.22", "1.23"),
     c("11111.222", "0.123"), c("11111.2222", "0.0123"), c("11111.2222200", "0.0000123"), c("11111.2222200000", "0.0000000123")),
     paste, collapse=paste("", .pm, "")))
-  expect_equal(format(x, notation="plus-minus", digits="pdg"), sapply(list(
-    c("10000", "12000000"), c("11100", "1200"), c("11111", "12"), c("11111.2", "1.2"),
-    c("11111.22", "0.12"), c("11111.222", "0.012"), c("11111.222220", "0.000012"), c("11111.222220000", "0.000000012")),
-    paste, collapse=paste("", .pm, "")))
```
Please restore

Enchufa2 commented on tests/testthat/test-print.R:
```diff
@@ -32,33 +29,27 @@ test_that("error formatting works properly", {
     c("(1.11112", "0.00001)e4"), c("(1.111122", "0.000001)e4"), c("(1.111122222", "0.000000001)e4"),
     c("(1.111122222000", "0.000000000001)e4")),
     paste, collapse=paste("", .pm, "")))
-
-  x <- set_errors(rep(0.827, 3), c(0.119, 0.367, 0.962))
-  expect_equal(format(x, notation="plus-minus", digits="pdg"), sapply(list(
-    c("0.83", "0.12"), c("0.8", "0.4"), c("1", "1")),
-    paste, collapse=paste("", .pm, "")))
```
Please restore

Enchufa2 commented on tests/testthat/test-print.R:
```diff
@@ -32,33 +29,27 @@ test_that("error formatting works properly", {
   expect_equal(format(x - set_errors(10)), "0(1)")
   expect_equal(format(x - x), "0(0)")
 
-  x <- set_errors(c(0.4, NA, NaN, Inf, -Inf))
```
Please restore

Enchufa2 commented on tests/testthat/test-print.R:
```diff
@@ -32,33 +29,27 @@ test_that("error formatting works properly", {
   expect_equal(format(x[2]), "NA(NA)")
   expect_equal(format(x[3]), "NaN(NaN)")
   expect_equal(format(x[4]), "Inf(Inf)")
-  expect_equal(format(x[5]), "-Inf(Inf)")
-
-  x <- set_errors(c(0e10, 1e12), 0.1e12)
-  expect_equal(format(x), c("0.0(1)e12", "1.0(1)e12"))
-  expect_equal(format(x, digits=2), c("0.00(10)e12", "1.00(10)e12"))
```
Please restore

Enchufa2 reviewed (changes requested):
Great, thanks for this. I see that some of the changes that I did in the last commits have been undone for some reason. Please restore them. Comments below.

alusiani commented:
Sorry for the changes, I should have checked more carefully. I was merging from an older revision I did. Part of the changes were related to using meld with omission of whitespace diffs.

I have now applied all requested changes, committed and pushed to github.  As far as I understand, that should be enough, please let me know if I also need to resubmit the pull request.

Enchufa2 commented:
Thanks! I renamed (and shortened) the new option, and added a new argument `decimals` to give it more visibility. I also refactored the code a little bit. Since you didn't give me push permission, I'm closing this one in favour of #61, I hope you don't mind.

alusiani commented:
> Thanks! I renamed (and shortened) the new option, and added a new argument `decimals` to give it more visibility. I also refactored the code a little bit. Since you didn't give me push permission, I'm closing this one in favour of #61, I hope you don't mind.

No problem! I am not an expert of pull requests and have not realized that a push permission was asked or required.

Enchufa2 commented:
It is not required, but it is convenient when, like in this case, the owner of the repo wants to add some adjustments to the PR.
